### PR TITLE
enlightenment.efl: 1.23.3 -> 1.24.0; new test module

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

- enlightenment.efl: format with nixpkgs-fmt
- enlightenment.efl: 1.23.3 -> [1.24.0](https://www.enlightenment.org/news/efl-1.24.0)
- enlightenment.efl: migrate to python3
- enlightenment.econnman: migrate to python3
- pythonPackages.pythonefl: format
- pythonPackages.python-efl: 1.23.0 -> [1.24.0](https://sourceforge.net/p/enlightenment/mailman/message/36997426/)

_Added_:
- nixos/tests: add enlightenment desktop environment

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by romildo** [comment]
@GrahamcOfBorg test enlightenment

File: pkgs/development/python-modules/python-efl/default.nix [hunk 1]
```diff
@@ -18,12 +24,16 @@ buildPythonPackage rec {
   propagatedBuildInputs = [ python.pkgs.dbus-python ];
```
Comment (jonringer) [inline pkgs/development/python-modules/python-efl/default.nix:24]:
```suggestion
  propagatedBuildInputs = [ dbus-python ];
```

File: pkgs/development/python-modules/python-efl/default.nix [hunk 2]
```diff
@@ -1,14 +1,20 @@
-{ stdenv, fetchurl, buildPythonPackage, pkgconfig, python, enlightenment }:
+{ stdenv
+, fetchurl
+, buildPythonPackage
+, pkgconfig
+, python
```
Comment (jonringer) [inline pkgs/development/python-modules/python-efl/default.nix:5]:
```suggestion
, python
, dbus-python
```

**Review by jonringer — COMMENTED** [review]
for `python-modules`, generally you will import the individual python package, instead of pulling them off of the `python.pkgs` attr. Allows for overriding that one package.

**Comment by jonringer** [comment]
The fixup should be squashed into the `pythonPackages.python-efl: 1.23.0 -> 1.24.0` commit

**Comment by romildo** [comment]
> The fixup should be squashed into the `pythonPackages.python-efl: 1.23.0 -> 1.24.0` commit

Done.

**Pull request merged**